test/python/algorithms/optimizers/test_umda.py

```diff
@@ -0,0 +1,97 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Tests for the UMDA optimizer."""
```

Comment:
In this file we test a bunch of optimizers against rosen - not sure how this would fare there https://github.com/Qiskit/qiskit-terra/blob/main/test/python/algorithms/optimizers/test_optimizers.py If it works it can be added.

Reply:
I have changed the tests and checked all without calling the private methods. I think all the needed tests are in test_umda.py. I would leave it as it is now.